spec/requests/vms_spec.rb

```diff
@@ -1703,6 +1703,64 @@ def query_match_regexp(*tables)
     end
   end
 
+  context "Vm rename action" do
+    it "to an invalid vm" do
+      api_basic_authorize action_identifier(:vms, :rename)
+
+      post(invalid_vm_url, :params => gen_request(:rename, "new_name" => "test"))
+
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it "to an invalid vm without appropriate role" do
+      api_basic_authorize
+
+      post(invalid_vm_url, :params => gen_request(:rename, "new_name" => "test"))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it "to a vm with missing new_name" do
+      api_basic_authorize action_identifier(:vms, :rename)
+
+      post(vm_url, :params => gen_request(:rename))
+
+      expect_bad_request("Must specify a new_name")
+    end
+
+    it "to a single vm" do
+      api_basic_authorize action_identifier(:vms, :rename)
+
+      post(vm_url, :params => gen_request(:rename, "new_name" => "new_vm"))
+
+      expect_single_action_result(:success => true, :message => /#{vm.id}.* renaming to new_vm/i, :href => api_vm_url(nil, vm))
```

Comment:
@lfu these probably require :task => true since we expect the response to include the task_href and task_id.